test/cpp/api/sequential.cpp

```diff
@@ -207,46 +208,46 @@ TEST_F(SequentialTest, AccessWithPtr) {
 TEST_F(SequentialTest, CallingForwardOnEmptySequentialIsDisallowed) {
   Sequential empty;
   ASSERT_THROWS_WITH(
-      empty->forward<int>(), "Cannot call forward() on an empty Sequential");
+      empty->forward(), "Cannot call forward() on an empty Sequential");
 }
 
 TEST_F(SequentialTest, CallingForwardChainsCorrectly) {
   struct MockModule : torch::nn::Module {
-    explicit MockModule(int value) : expected(value) {}
-    int expected;
-    int forward(int value) {
-      assert(value == expected);
+    explicit MockModule(int value) : expected(torch::tensor(value)) {}
+    torch::Tensor expected;
+    torch::Tensor forward(torch::Tensor value) {
+      assert((value == expected).item<bool>());
       return value + 1;
     }
   };
 
   Sequential sequential(MockModule{1}, MockModule{2}, MockModule{3});
 
-  ASSERT_EQ(sequential->forward<int>(1), 4);
+  ASSERT_TRUE((sequential->forward(torch::tensor(1)), torch::tensor(4)).item<float>());
```

Comment:
We can change this to `ASSERT_TRUE(sequential->forward(torch::tensor(1)).equal(torch::tensor(4)))`, to avoid having to call `.item<float>()`.